src/transformers/models/beit/image_processing_beit.py

```diff
@@ -0,0 +1,519 @@
+            not included in all classes of a dataset (e.g. ADE20k). The background label will be replaced by 255.
+    """
+
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self,
+        do_resize: bool = True,
+        size: Dict[str, int] = None,
+        resample: PILImageResampling = PILImageResampling.BICUBIC,
+        do_center_crop: bool = True,
+        crop_size: Dict[str, int] = None,
+        rescale_factor: Union[int, float] = 1 / 255,
+        do_rescale: bool = True,
+        do_normalize: bool = True,
+        image_mean: Optional[Union[float, List[float]]] = None,
+        image_std: Optional[Union[float, List[float]]] = None,
+        do_reduce_labels: bool = False,
+        **kwargs
+    ) -> None:
+        if "reduce_labels" in kwargs:
+            warnings.warn(
+                "The `reduce_labels` parameter is deprecated and will be removed in a future version. Please use"
+                " `do_reduce_labels` instead.",
+                FutureWarning,
+            )
+            do_reduce_labels = kwargs.pop("reduce_labels")
+        super().__init__(**kwargs)
+        size = size if size is not None else {"height": 256, "width": 256}
+        size = get_size_dict(size)
+        crop_size = crop_size if crop_size is not None else {"height": 224, "width": 224}
+        crop_size = get_size_dict(crop_size)
+        self.do_resize = do_resize
+        self.size = size
+        self.resample = resample
+        self.do_center_crop = do_center_crop
+        self.crop_size = crop_size
+        self.do_rescale = do_rescale
+        self.rescale_factor = rescale_factor
+        self.do_normalize = do_normalize
+        self.image_mean = image_mean if image_mean is not None else IMAGENET_STANDARD_MEAN
+        self.image_std = image_std if image_std is not None else IMAGENET_STANDARD_STD
+        self.do_reduce_labels = do_reduce_labels
+
+    def resize(
+        self,
+        image: np.ndarray,
+        size: Dict[str, int],
+        resample: PILImageResampling = PILImageResampling.BICUBIC,
+        data_format: Optional[Union[str, ChannelDimension]] = None,
+        **kwargs
+    ) -> np.ndarray:
+        """
+        Resize an image to (size["height"], size["width"]).
+
+        Args:
+            image (`np.ndarray`):
+                Image to resize.
+            size (`Dict[str, int]`):
+                Size of the output image.
+            resample (`PILImageResampling`, *optional*, defaults to `PIL.Image.BICUBIC`):
+                Resampling filter to use when resiizing the image.
+            data_format (`str` or `ChannelDimension`, *optional*):
+                The channel dimension format of the image. If not provided, it will be the same as the input image.
+        """
+        size = get_size_dict(size)
+        return resize(
+            image, size=(size["height"], size["width"]), resample=resample, data_format=data_format, **kwargs
+        )
```

Comment:
The get_size_dict function possibly converts the size to a dict containing only "shortest_edge". So the resize will fail in that case as it's expecting both "height" and "width" keys, correct?

Reply:
However, the default value will of course work. Just wondering whether we should take into account the possibility.

Reply:
For this image processor, it would only convert to `{"shortest_edge": ...}` if the input was `{"shortest_edge": ...}`, in which case you're right it would fail. In the method docstrings, it does specify the type of dictionary expected e.g. `{"height": ..., "width": ...}`. I'll add a check, raising a ValueError if the input is wrong. 

I think this is the safest and cleanest way to handle the different forms `size` as this image processor doesn't have to know about the behaviour of any other image processor e.g. if one image processor needed a new key for `size` all the other image processors wouldn't need to update to accept this. In particular, I don't think we should add support for accepting `{"shortest_edge": ...}` for all the resize methods as this is a PyTorch specific behaviour and not explicit whereas `{"height":..., "width"...}` being universally accepted could be done.

Reply:
I've added exceptions, so that an image processors `resize` method will fail without the expected input format.